crates/codegen/syntax_templates/src/rust/parse_output.rs

```diff
@@ -0,0 +1,51 @@
+use std::{collections::BTreeSet, ops::Range};
+
+use super::{cst, kinds::TokenKind, parse_error::render_error_report};
+
+#[derive(Debug, PartialEq)]
+pub struct ParseOutput {
+    pub(crate) parse_tree: cst::Node,
+    pub(crate) errors: Vec<ParseError>,
+}
+
+impl ParseOutput {
+    pub fn parse_tree(&self) -> cst::Node {
+        return self.parse_tree.clone();
+    }
+
+    pub fn errors(&self) -> &Vec<ParseError> {
+        return &self.errors;
+    }
+
+    pub fn is_valid(&self) -> bool {
+        return self.errors.is_empty();
+    }
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct ParseError {
+    pub(crate) utf8_range: Range<usize>,
```

Comment:
I wonder why are we not using a `TextIndex` here. Since we expose this information to the public API, shouldn't we expose the other indices as well?

Reply:
done